mne_bids/path.py

```diff
@@ -483,15 +483,33 @@ def update(self, *, check=None, **kwargs):
         >>> print(bids_path.basename)
         sub-test_ses-two_acq-test_ieeg.vhdr
         """
-        run = kwargs.get('run')
-        if run is not None and not isinstance(run, str):
-            # Ensure that run is a string
-            kwargs['run'] = '{:02}'.format(run)
+        for key, val in kwargs.items():
+            if key == 'root':
+                _validate_type(val, types=('path-like', None), item_name=key)
+                continue
+
+            if key == 'datatype':
+                if val is not None and val not in ALLOWED_DATATYPES:
+                    raise ValueError(f'datatype ({val}) is not valid. '
+                                     f'Should be one of '
+                                     f'{ALLOWED_DATATYPES}')
+                else:
+                    continue
+
+            if key not in ENTITY_INDEX_OR_LABEL:
+                raise ValueError(f'Key must be one of '
+                                 f'{ALLOWED_PATH_ENTITIES}, got {key}')
 
-        split = kwargs.get('split')
-        if split is not None and not isinstance(split, str):
-            # Ensure that run is a string
-            kwargs['split'] = '{:02}'.format(split)
+            if ENTITY_INDEX_OR_LABEL[key] == 'label':
+                _validate_type(val, types=(None, str),
+                               item_name=key)
+            else:
+                assert ENTITY_INDEX_OR_LABEL[key] == 'index'
+                _validate_type(val, types=(int, str, None), item_name=key)
+                if isinstance(val, str) and not val.isdigit():
+                    raise ValueError(f'{key} is not an index (Got {val})')
+                elif isinstance(val, int):
+                    kwargs[key] = '{:02}'.format(val)
```

Comment:
Since this would only apply to `split` and `run`, if I'm not mistaken, I'm not sure we should zero-pad here at all, as but indices in most cases be <10, no?

Reply:
I thought the same, however *sometimes* they may be over 10 ... while they will be probably never > 100

so zero padding to length 2 covers *all* cases, at the expense that those with <10 will see zero-padded indices (I personally would not care about that)

so for me it's +1 to zero pad

Reply:
Ok, good point.

Reply:
this was already done this way. And it's also safe for numbers > 99

In [1]: '{:02}'.format(1)
Out[1]: '01'

In [2]: '{:02}'.format(100)
Out[2]: '100'

In [3]: '{:02}'.format(10)
Out[3]: '10'